src/use_case/add_exercise/AddExerciseDataAccessInterface.java

```diff
@@ -1,5 +1,12 @@
 package use_case.add_exercise;
 
+import java.util.ArrayList;
+
 public interface AddExerciseDataAccessInterface {
+    boolean existsById(int id);
+
+    boolean existsByName(String exercise_name);
```

Comment:
have questions abt this, check discord

Reply:
> have questions abt this, check discord

I was thinking check to make sure the routine exists, then check the desired exercise exists. Does that make sense with the implementation.

Reply:
on discord i asked how we're going to get the String exercise_name. your input data assumes this use case will just be fed an exercise name (the only way this function works is if you can get that exercise_name is via look up beforehand somehow) so i think my suggestion is either to change these to two methods: getNameByTarget (returns a string) and getName (also returns a string) and your input data would be a query string "target" or "name" so the DAO can go searching for your ercise_name (this is how I'm writing the look up use case) OR, we implement look up to be used WITH your use case, so the look up will provide you with your exercise (in that case, it could just send you an exercise entity, instead of an exercise_name). i think communication on discord may be better for discussing this in detial.